src/functional/ffront/ast_passes/simple_assign.py

```diff
@@ -67,17 +67,35 @@ class SingleAssignTargetPass(NodeYielder):
     ...     )
     ... ))
     def foo():
-        a = 1
-        b = 1
+        __sat_tmp0 = 1
+        a = __sat_tmp0
+        b = __sat_tmp0
         return (a, b)
 
     """
 
+    unique_name_id: int = 0
+
+    def _unique_symbol_name(self) -> ast.Name:
+        name = ast.Name(id=f"__sat_tmp{self.unique_name_id}", ctx=ast.Store())
+        self.unique_name_id += 1
+        return name
```

Comment:
I guess it works for now, but I feel like it might be worth considering having some sort of infrastructure to generate fresh variable names (that are guaranteed not to be used anywhere else) and maybe have some sort of way to distinguish between user-provided names and newly generated names.